src/icons/YouTubeIcon.tsx

```diff
@@ -1,14 +1,13 @@
 import React from 'react';
-import { IconProps } from '../Contract';
+import { IconPropsWithColor } from '../Contract';
 
-export default function YouTubeIcon(props: IconProps) {
-    const width = props.width ?? 33;
-    const height = props.height ?? 33;
-    const color = props.color ?? 'white';
-    const secondaryColor = props.secondaryColor ?? 'black';
+export default function YouTubeIcon(props: IconPropsWithColor) {
+    const size = props.size ?? 33;
+    const color = props.color?.primaryColor ?? 'white';
+    const secondaryColor = props.color?.secondaryColor ?? 'black';
```

Comment:
Also I thought that values of attributes of each icon component (e.g. size, color) after ?? operator can be used as default values - so size and color props are only used when creating icon different than default.

Reply:
Yes, nullable operator is a good approach for default values